src/main/java/sirius/biz/storage/layer2/mongo/MongoProcessBlobChangesLoop.java

```diff
@@ -78,4 +78,45 @@ protected void propagateDelete(@Nonnull Directory dir) {
              .executeFor(MongoDirectory.class);
         mongo.update().set(MongoBlob.DELETED, true).where(MongoBlob.PARENT, directoryId).executeFor(MongoBlob.class);
     }
+
+    @Override
+    protected void processRenamedDirectories(Runnable counter) {
+        mango.select(MongoDirectory.class).eq(MongoDirectory.RENAMED, true).limit(256).iterateAll(dir -> {
+            try {
+                propagateRename(dir);
+                mongo.update()
+                     .set(MongoDirectory.RENAMED, false)
+                     .where(MongoDirectory.ID, dir.getId())
+                     .executeFor(MongoDirectory.class);
+            } catch (Exception e) {
+                handleDirectoryRenameException(dir, e);
+            }
+            counter.run();
+        });
+    }
+
+    @Override
+    protected void propagateRename(@Nonnull Directory dir) {
+        String directoryId = dir.getIdAsString();
+        mongo.update()
+             .set(MongoDirectory.RENAMED, true)
+             .where(MongoDirectory.PARENT, directoryId)
+             .executeFor(MongoDirectory.class);
+        mongo.update()
+             .set(MongoBlob.PARENT_CHANGED, true)
+             .where(MongoBlob.PARENT, directoryId)
+             .executeFor(MongoBlob.class);
+    }
+
+    @Override
+    protected void processParentChangedBlobs(Runnable counter) {
+        mango.select(MongoBlob.class).eq(MongoBlob.PARENT_CHANGED, true).limit(256).iterateAll(blob -> {
```

Comment:
didn't we want to increase all the 256's to s.th. larger - like 1k oder 512? goes for all blocks. First step would be to define a constant in a base class and use this everywhere to it can be centrally controlled(?)